var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -227,6 +226,8 @@ class Petsc(Package):
     depends_on('superlu-dist@develop+int64', when='@develop+superlu-dist+mpi+int64')
     depends_on('mumps+mpi', when='+mumps+mpi~int64')
     depends_on('scalapack', when='+mumps+mpi~int64')
+    depends_on('mumps+mpi', when='@3.13:+mumps+mpi')
+    depends_on('scalapack', when='@3.13:+mumps+mpi')
```

Comment:
I think the current code is buggy. It just ignores the variant combinations that don't work. So perhaps its better to use conflicts here. [and there is already a  `conflicts('+mumps', when='~mpi')`]

How about:
```
-    depends_on('mumps+mpi', when='+mumps+mpi~int64')
-    depends_on('scalapack', when='+mumps+mpi~int64')
+    depends_on('mumps+mpi', when='+mumps+mpi')
+    depends_on('scalapack', when='+mumps+mpi')
+    conflicts('+mumps', when='@:3.12.99+mpi+int64')
```

Reply:
Ops - I meant to say:
```
-    depends_on('mumps+mpi', when='+mumps+mpi~int64')
-    depends_on('scalapack', when='+mumps+mpi~int64')
-    depends_on('mumps+mpi', when='@3.13:+mumps+mpi')
-    depends_on('scalapack', when='@3.13:+mumps+mpi')
+    conflicts('+mumps', when='@:3.12.99+int64')
+    depends_on('mumps', when='+mumps')
+    depends_on('scalapack', when='+mumps')
```

Reply:
Yes !
With the current code, spack install petsc@3.12 +mumps +int64 just installs petsc +int64 _without mumps_.
Since I am not very familiar with writing packages, I tried to keep existing (albeit surprising) behavior, and just add the new functionality, but your suggestion is much better.
I'm also quite new with github, should I do something to apply your suggestion and resubmit ?

Reply:
How about 
```
-    depends_on('mumps+mpi', when='+mumps+mpi~int64')
-    depends_on('scalapack', when='+mumps+mpi~int64')
-    depends_on('mumps+mpi', when='@3.13:+mumps+mpi')
-    depends_on('scalapack', when='@3.13:+mumps+mpi')
+    conflicts('+mumps', when='@:3.12.99+int64')
+    depends_on('mumps~int64', when='+mumps')
+    depends_on('scalapack', when='+mumps')
```
Because a user might want to specify `spack install petsc+mumps+int64 ^mumps+int64`, which as far as I understand should not work.